src/web/client/services/etagHandlerService.ts

```diff
@@ -134,18 +137,83 @@ export class EtagHandlerService {
                     new Date().getTime() - requestSentAtTime,
                     authError,
                     '',
-                    telemetryEventNames.WEB_EXTENSION_ETAGHANDLERSERVICE_API_ERROR,
-                    (error as Response)?.status as unknown as string
+                    telemetryEventNames.WEB_EXTENSION_ETAG_HANDLER_SERVICE_API_ERROR,
+                    (error as Response)?.status.toString()
                 );
-            }else{
+            } else {
                 WebExtensionContext.telemetry.sendErrorTelemetry(
-                    telemetryEventNames.WEB_EXTENSION_ETAGHANDLERSERVICE_ERROR,
+                    telemetryEventNames.WEB_EXTENSION_ETAG_HANDLER_SERVICE_ERROR,
                     (error as Error)?.message
                 );
             }
-            
         }
 
         return "";
     }
+
```

Comment:
The fetch and etag update part can be merged with the function above and will add more telemetry in logs for this event as well in next PR